### ISSUE TITLE
Monitors connected through the dock are blank on Wayland but work fine on X11 (Ubuntu 24.04)

### ISSUE THREAD

**Issue by zjeffer** [DisplayLink/evdi#484]
* Are you using the latest driver? **Yes**
    
  * Are you using the latest EVDI version? **Yes**
    
  * If you are using a DisplayLink device, have you checked 'troubleshooting'
    on DisplayLink's website? **Yes**
    
  * Is this issue related to evdi/kernel? **Yes**
    

  * Linux distribution and its version: **Ubuntu 24.04**
  * Linux kernel version: `6.8.0-40-generic`
  * Xorg version (if used) `7.7+23ubuntu3`, core: `21.1.12-1ubuntu1`
  * Desktop environment in use: `gnome 46.0`
  * GPU: Nvidia RTX 3070 Laptop
  * GPUDriver: `nvidia-550` (proprietary)
  * CPU: AMD Ryzen 7 5800H

I have a Dell D6000 docking station connected to my Lenovo Legion 5 Pro laptop. Everything works (keyboard & mouse through usb, ethernet), but my monitors do not display anything on Wayland. Connecting the cable turns on the monitor but then it goes to standby because it gets 'No signal'.

On X11, everything works fine. On Windows too. 

* `dmesg` logs when reconnecting an HDMI 1920x1080@60 monitor to my dock on Wayland, with evdi verbose logging (level 6) (disconnecting, waiting a couple seconds, then reconnecting again): [evdi-wayland.log](https://github.com/user-attachments/files/16570289/evdi-wayland.log). Reconnection happens between the empty lines I added.

* `dmesg` logs when reconnecting (same thing as above) on X11: [evdi-xorg.log](https://github.com/user-attachments/files/16570278/evdi-xorg.log). I left some newlines on lines 63-67 to show where I reconnected.

* Xrandr output on Xorg, with the monitor connected (DVI-I-1-1): [xrandr-xorg.log](https://github.com/user-attachments/files/16570301/xrandr-xorg.log). 

I compared the logs and noticed that on Wayland, after the Edid property is set, it immediately logs `evdi_user_framebuffer_destroy`, `evdi_painter_close`, and `evdi_driver_close`. These logs don't occur on X when connecting the monitor.

---

I also notice that when I use the nouveau drivers instead of the proprietary Nvidia drivers, the monitor does get a signal, but only the cursor, the background is black and I can't move windows to the monitor. Either way, using nouveau isn't an option, as I'm planning to use this dock on Hyprland, and play games: both of which aren't great with nouveau. But this info might be useful to determine the cause.

Let me know if you need any other logs.

**Comment by zjeffer** [comment]
On Hyprland, reconnecting a monitor has the same behaviour but different logs: [evdi-hyprland.log](https://github.com/user-attachments/files/16575501/evdi-hyprland.log)

**Comment by zjeffer** [comment]
Seems it happens due to the combination Nvidia + Wayland: I see the same behaviour on KDE + Wayland and on Hyprland when using Nvidia, but when running them on my Intel Integrated GPU it works fine.

Is this the right place to report this? Or should I report it to the nvidia forums?

**Comment by zjeffer** [comment]
Still doesn't work with nvidia-560 drivers.

**Comment by zjeffer** [comment]
Created a thread in the nvidia forums here: https://forums.developer.nvidia.com/t/displaylink-monitor-doesnt-get-signal-when-connected-through-docking-station-on-nvidia-wayland/304676

**Comment by honeyraptor** [comment]
@zjeffer 
Were you able to resolve it?

I too am facing the same issue (Blank screen),

With 
Ubuntu 24.04.1 LTS
Nvidia 550.120
Wayland.
Gnome 46
DisplayLink - displaylink-driver/stable, 6.1.0-17 amd64


----------------------
However the setup works well,
 with X11.

**Comment by zjeffer** [comment]
No, still doesn't work. Tried again a few days ago with the latest evdi git commit, no difference. I'm on `nvidia-565.77`.

**Comment by zjeffer** [comment]
Good to hear I'm not alone with this issue :)
Which docking station do you use?

**Comment by honeyraptor** [comment]
@zjeffer 
I am using Alogic Universal Twin HD Pro.

**Comment by Pitastic** [comment]
Confirm it is working 🎉 on `1.14.14` ! (...could be closed from my side)

**Comment by zjeffer** [comment]
@Shakthi-Bala get out of here with that AI slop, that's just plain spam.

@Pitastic thanks for letting me know it's now working in a newer version. I haven't had time to test it yet, but I'll close it for now. Once I test it and I find it's not working, I'll reopen.
